### PR TITLE
Generate is Methods for boolean properties.

### PR DESCRIPTION
**Pull request by fattouchsquall**

### CODE REVIEW

**Comment by saimaz** [comment]
Can you rebase PR @fattouchsquall ? It should be open to 5.0 branch instead of master.

**Comment by fattouchsquall** [comment]
@saimaz it's done.

File: composer.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
     "minimum-stability": "dev",
     "extra": {
         "branch-alias": {
-            "dev-master": "5.0-dev"
+            "dev-master": "5.1-dev"
```
Comment (saimaz) [inline composer.json]: After rebase, it should be `5.0-dev`

File: Generator/DocumentGenerator.php [hunk 2]
```diff
@@ -141,7 +157,15 @@ private function generateDocumentMethods(array $metadata)
 
         foreach ($metadata['properties'] as $property) {
             $lines[] = $this->generateDocumentMethod($property, $this->setMethodTemplate) . "\n";
-            $lines[] = $this->generateDocumentMethod($property, $this->getMethodTemplate) . "\n";
+            if (isset($property['property_type'])) {
```
Comment (saimaz) [inline Generator/DocumentGenerator.php]:
```       
     if (isset($property['property_type']) && $property['property_type'] === 'boolean') {
                    $lines[] = $this->generateDocumentMethod($property, $this->isMethodTemplate) . "\n";
            } else {
                $lines[] = $this->generateDocumentMethod($property, $this->getMethodTemplate) . "\n";
            }
```

Is the same? Not?

Comment (saimaz) [inline Generator/DocumentGenerator.php]: And for isser we might want to add getter as well.

Comment (saimaz) [inline Generator/DocumentGenerator.php]: So:
```
if (isset($property['property_type']) && $property['property_type'] === 'boolean') {
                    $lines[] = $this->generateDocumentMethod($property, $this->isMethodTemplate) . "\n";
            }
            $lines[] = $this->generateDocumentMethod($property, $this->getMethodTemplate) . "\n";
```

This is how it should look like.

File: Mapping/MetadataCollector.php [hunk 3]
```diff
@@ -112,7 +112,7 @@ public function getBundleMapping($name, $config = [])
 
         $this->enableCache && $mappings = $this->cache->fetch($cacheName);
 
-        if (isset($mappings) && false !== $mappings) {
+        if (isset($mappings) && $mappings) {
```
Comment (saimaz) [inline Mapping/MetadataCollector.php]: This doesn't belong here. After correct rebase it should disappear.

**Pull request closed without merging**